packages/polyserve/src/compile-middleware.ts

```diff
@@ -38,6 +38,45 @@ const compileMimeTypes = [htmlMimeType, ...javaScriptMimeTypes];
 // but not their tests.
 export const isPolyfill = /(^|\/)webcomponentsjs\/[^\/]+$/;
 
+export interface NPMPackage {
+  /**
+   * Name of the node package. e.g. '@polymer/polymer'
+   */
+  name: string;
+
+  /**
+   * JS entrypoints relative to packageName e.g. lodash/index.js would simply
+   * be ['index.js'] and myPackage/dist/addon.js and myPackage/lib/core.js would
+   * be ['dist/addon.js', 'lib/core.js']
+   */
+  jsEntrypoint: string;
+}
+
+// scripts to be injected into the running test
+export const ENVIRONMENT_SCRIPTS: NPMPackage[] = [
```

Comment:
What if we checked for a URL parameter like ?nocompile, and have WCT add that to the URLs it injects? Maybe that separates concerns better?

Reply:
Oooh I like that solution. Will take a stab at that 👍

Reply:
+1 to not adding more WCT-specific content directly to polyserve.

Reply:
Updated :+1: I will update the WCT stuff accordingly in the other PR to minimize noise.